Add Azure Blob Storage hosting guide and update local hosting docs (#1384)
nginx/documentation: content/nginxaas-azure/quickstart/hosting-static-content-blob-storage.md

---
title: Hosting static content in Azure Blob Storage
weight: 210
toc: true

> nd-docs: DOCS-1344

Reviewer: ```suggestion
```
Avoid duplicated UIDs

url: /nginxaas/azure/quickstart/hosting-static-content-blob-storage/
type:
- how-to
---

F5 NGINXaaS for Azure (NGINXaaS) can serve static content stored in Azure Blob Storage, allowing you to host large static websites without the configuration payload size limitations of local hosting. This approach also keeps your storage account private by restricting access to your NGINXaaS deployment.

## Prerequisites

- An Azure Storage Account
- An NGINXaaS for Azure deployment

> - Static content files to serve

Reviewer: Ideally these needs to be linked to relevant documentation (ours or Azure's)

## Configure Azure Blob Storage

### Step 1: Upload static files to the $web container

Upload your static files to the `$web` container in your storage account. This is the standard container used for static website hosting in Azure.

### Step 2: Configure network access

1. In your storage account, navigate to **Networking** under **Security + networking**.
2. Under **Public network access**, select **Enable public access from selected virtual networks and IP addresses**.
3. In the **Virtual networks** section, click **Add existing virtual network**.
4. Select the virtual network and subnet where your NGINXaaS deployment is located.
5. Click **Add** to allow your NGINXaaS deployment to connect to the storage account.

{{< call-out "note" >}}This configuration ensures that your storage account is not accessible from the public Internet, only from your NGINXaaS deployment's subnet.{{< /call-out >}}

> ### Step 3: Enable static website hosting

Reviewer: ```suggestion
### Enable static website hosting
```

1. In your storage account, navigate to **Static website** under **Data management**.
2. Enable **Static website**.
3. Set your **Index document name** (e.g., `index.html`).
4. Optionally, set an **Error document path** for 404 errors.
5. Click **Save**.

Note the **Primary endpoint** URL that appears after enabling static website hosting. You'll use this URL in your NGINX configuration.

## Configure NGINXaaS

Create an NGINX configuration that proxies requests to your Azure Blob Storage static website endpoint:

```nginx
user nginx;
worker_processes auto;
worker_rlimit_nofile 8192;
pid /run/nginx/nginx.pid;

error_log /var/log/nginx/error.log error;

http {
    upstream storage_origin {
        server your-storage-account.z20.web.core.windows.net:443;
        keepalive 32;
    }

    server {
        listen 443 ssl;
        ssl_certificate /etc/nginx/example.cert;
        ssl_certificate_key /etc/nginx/example.key;

        location /static/ {
            proxy_pass https://storage_origin/content/;
            proxy_set_header Host your-storage-account.z20.web.core.windows.net;
            proxy_http_version 1.1;
            proxy_set_header Connection "";
        }
    }
}
```

> {{< call-out "important" >}}Replace `your-storage-account` with your actual storage account name in both the upstream server definition and the `proxy_set_header Host` directive. The region code (e.g., `z20`) may vary depending on your storage account's region.{{< /call-out >}}

Reviewer: ```suggestion
{{< call-out "important" >}}Replace `your-storage-account` with your actual storage account name in both the upstream server definition and the `proxy_set_header Host` directive. The region code (for example, `z20`) may vary depending on your storage account's region.{{< /call-out >}}
```

avoid latin terms like e.g., etc...
